cms/djangoapps/contentstore/core/tests/test_course_optimizer_provider.py

```diff
@@ -0,0 +1,190 @@
+"""
+Tests for course optimizer
+"""
+
+import unittest
+from unittest.mock import Mock, patch
+
+from cms.djangoapps.contentstore.tests.utils import CourseTestCase
+from cms.djangoapps.contentstore.core.course_optimizer_provider import (
+    generate_broken_links_descriptor,
+    _update_node_tree_and_dictionary,
+    _get_node_path,
+    _create_dto_from_node_tree_recursive
+)
+
+class TestLinkCheck(CourseTestCase):
+    """
+    Tests for the link check functionality
+    """
+    def setUp(self):
+        """Setup for tests"""
+        global MOCK_TREE
+        global MOCK_XBLOCK_DICTIONARY
+        global MOCK_COURSE
+        global MOCK_SECTION
+        global MOCK_SUBSECTION
+        global MOCK_UNIT
+        global MOCK_BLOCK
+
+        MOCK_TREE = {
+            'chapter_1': {
+                'sequential_1': {
+                    'vertical_1': {
+                        'block_1': {}
+                    }
+                }
+            }
+        }
+
+        MOCK_XBLOCK_DICTIONARY = {
+            'chapter_1': {
+                'display_name': 'Section Name',
+                'category': 'chapter'
+            },
+            'sequential_1': {
+                'display_name': 'Subsection Name',
+                'category': 'sequential'
+            },
+            'vertical_1': {
+                'display_name': 'Unit Name',
+                'category': 'vertical'
+            },
+            'block_1': {
+                'display_name': 'Block Name',
+                'url': '/block/1',
+                'broken_links': ['broken_link_1', 'broken_link_2'],
+                'locked_links': ['locked_link']
+            }
+        }
+
+        MOCK_COURSE = Mock()
+        MOCK_SECTION = Mock(
+            location=Mock(block_id='chapter_1'),
+            display_name='Section Name',
+            category='chapter'
+        )
+        MOCK_SUBSECTION = Mock(
+            location=Mock(block_id='sequential_1'),
+            display_name='Subsection Name',
+            category='sequential'
+        )
+        MOCK_UNIT = Mock(
+            location=Mock(block_id='vertical_1'),
+            display_name='Unit Name',
+            category='vertical'
+        )
+        MOCK_BLOCK = Mock(
+            location=Mock(block_id='block_1'),
+            display_name='Block Name',
+            course_id='course-v1:test+course',
+            category='html'
+        )
+        # MOCK_BLOCK.location.__str__.return_value = 'block_location_str'
+        MOCK_COURSE.get_parent.return_value = None
+        MOCK_SECTION.get_parent.return_value = MOCK_COURSE
+        MOCK_SUBSECTION.get_parent.return_value = MOCK_SECTION
+        MOCK_UNIT.get_parent.return_value = MOCK_SUBSECTION
+        MOCK_BLOCK.get_parent.return_value = MOCK_UNIT
+
+
+    def test_update_node_tree_and_dictionary(self):
```

Comment:
Can you have a look at https://github.com/openedx/frontend-lib-content-components/blob/main/docs/decisions/0006-test-names.rst and follow the naming conventions there?

Reply:
Also the convention we use for pytest is that the function docstring describes quite clearly what the unit under test is supposed to do and what we are testing.